prompt2model/dataset_retriever/dataset_index_file/retrieve_dataset_info_with_configs.py

```diff
@@ -0,0 +1,285 @@
+from __future__ import annotations  # noqa FI58
+
+import argparse
+import gc
+import json
+import multiprocessing
+import sys
+import threading
+import time
+from collections.abc import MutableMapping
+from pathlib import Path
+from typing import Any
+
+import datasets
+import requests
+
+EXCLUDED_TAGS = [
+    "arxiv",
+    "region",
+    "license",
+    "size_categories",
+    "language_creators",
+]
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--index", type=int, default=100)
+
+parser.add_argument("--num_processes", type=int, default=4)
```

Comment:
In our meeting, you mentioned that you didn't use multi-processing to create the index. Could all of the multi-processing code be removed?